### PR TITLE
FilenameBear.py: Add prefix/suffix matching feature

### PR DESCRIPTION
**Pull request by AMR-KELEG**
FilenameBearTest.py: Add tests

Add a new feature to the FilenameBear.
The bear can check if the filename matches a certain prefix/suffix.

Closes https://github.com/coala/coala-bears/issues/1817

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/general/FilenameBear.py [hunk 1]
```diff
@@ -71,6 +77,18 @@ def run(self, filename, file,
...
+            yield Result.from_values(
+                origin=self,
+                message='Filename does not use the {} prefix.'.format(prefix),
+                file=filename)
+
+        if not filename_without_extension.endswith(suffix):
+            yield Result.from_values(
+                origin=self,
+                message='Filename does not use the {} suffix.'.format(suffix),
+                file=filename)
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: Is it possible to suggest a patch?

File: bears/general/FilenameBear.py [hunk 2]
```diff
@@ -27,7 +27,9 @@ class FilenameBear(LocalBear):
 
     def run(self, filename, file,
             file_naming_convention: str=None,
-            ignore_uppercase_filenames: bool=True):
+            ignore_uppercase_filenames: bool=True,
+            prefix: str='',
+            suffix: str=''):
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: I would call those settings `filename_prefix` and `filename_suffix` (or `file_name_prefix` / `file_name_suffix`, what you like better). There are certainly other prefixes and suffixes that mean something different in another context^^

File: bears/general/FilenameBear.py [hunk 3]
```diff
@@ -43,6 +45,10 @@ def run(self, filename, file,
         :param ignore_uppercase_filenames:
             Whether or not to ignore fully uppercase filenames completely,
             e.g. COPYING, LICENSE etc.
+        :param prefix:
+            Check whether the filename uses a certain prefix.
+        :param suffix:
+            Check whether the filename uses a certain suffix.
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: You should definitely mention that the file ending is not checked. Otherwise I would have though that I need to include the file-ending^^ (as a filename consists of `name.ending`)

File: bears/general/FilenameBear.py (hunk 1 above)
Comment (Makman2) [inline bears/general/FilenameBear.py]: Like just appending the suffix^^
(and for the prefix: just appending the prefix ;D)

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Do you mean that the bear should be able to rename the file such that the file name matches the prefix/suffix?

File: bears/general/FilenameBear.py (hunk 3 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: I thought that this is obvious especially for the suffix part.
Anyway, i will update the description.

File: bears/general/FilenameBear.py (hunk 1 above)
Comment (Makman2) [inline bears/general/FilenameBear.py]: If you suggest patches, we maybe need to take care if a file has not both: correct prefix and suffix. Then we would get a merge conflict for the patches^^ We could merge the results into a single result and merge the messages together (depending what combination we get).

Comment (Makman2) [inline bears/general/FilenameBear.py]: oh we even need to merge it below with https://github.com/coala/coala-bears/pull/1836/commits/23360b43da416d3afae8573e646d24ecb83ae6fe#diff-8da09498fa5dc8583a01d5e2f76de875R98 :/

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Ok, i will dig dipper into the Result class and try to figure out how to suggest patches.
Thanks for your comments.

Comment (Makman2) [inline bears/general/FilenameBear.py]: See https://github.com/coala/coala/blob/master/coalib/results/Diff.py :) There's a `rename` field you can use for this purpose ;)

File: bears/general/FilenameBear.py [hunk 4]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
+                                         if
+                                         filename_without_extension.endswith(
+                                                               filename_suffix)
+                                         else
+                                         correct_prefix_name + filename_suffix)
+
+            diff = Diff(file, rename=os.path.join(
+                head, correct_prefixsuffix_name + extension))
+
+            FilenameBear_messages = [
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: why did you prefix with `FilenameBear`? I think just `messages` is totally fine :)

File: bears/general/FilenameBear.py [hunk 5]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
+                'Filename does not follow {} naming-convention.'.format(
+                        file_naming_convention),
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: we usually use +4 spaces for hanging indents instead of 8^^

File: bears/general/FilenameBear.py [hunk 6]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
+                        filename_prefix),
+                'Filename does not use the {} suffix.'.format(
+                        filename_suffix)]
+
+            selectors = [
+                        new_name != filename_without_extension,
+                        not filename_without_extension.startswith(
+                            filename_prefix),
+                        not filename_without_extension.endswith(
+                            filename_suffix)]
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: indentation: 4 spaces relative to `selectors`:
```
selectors = [
    new_name != filename_without_extensionm
    not filename_without_extension.startswith(
        filename_prefix),
    not ...
```

File: bears/general/FilenameBear.py [hunk 7]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
+                                                               filename_prefix)
+                                   else
+                                   filename_prefix + new_name)
+
+            correct_prefixsuffix_name = (correct_prefix_name
+                                         if
+                                         filename_without_extension.endswith(
+                                                               filename_suffix)
+                                         else
+                                         correct_prefix_name + filename_suffix)
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: why you don't move the new filename generation before the `if` clause? So you modify `new_name` before the if, and in the end you don't have to check every condition, but just like before `new_name != filename_without_extension`

Comment (Makman2) [inline bears/general/FilenameBear.py]: So into this line: https://github.com/coala/coala-bears/pull/1836/commits/9c4d2ed9295e8d067f7d60fe7210ae6344f4b027#diff-8da09498fa5dc8583a01d5e2f76de875R85

File: bears/general/FilenameBear.py [hunk 8]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        if (new_name != filename_without_extension or
+            not filename_without_extension.startswith(filename_prefix) or
+                not filename_without_extension.endswith(filename_suffix)):
+            correct_prefix_name = (new_name
+                                   if
+                                   filename_without_extension.startswith(
+                                                               filename_prefix)
+                                   else
+                                   filename_prefix + new_name)
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: Should be a new issue, but I think I got a smart idea :P

Instead of just prepending the prefix if we don't match it in the beginning, we try to match it as long as possible. I.e. consider this:
A file is called `hello_world.py`. We want it to have the prefix `python_` (whyever...). Our patch would suggest to correct it to `python_hello_world`. This is fine. But now consider a filename with a typo:
`pythonhello_world2.py`. Our bear would propose now `python_pythonhello_world2.py`. A human would see that we probably meant to type `python_hello_world2.py`. Same with this file: `ython_hello_world3.py` is probably meant to be `python_hello_world3.py`.

Like said this is imo a new issue, but we could try to perform some best-suiting match of the prefix (same idea would also apply for suffixes) and suggest a smarter patch :)

File: bears/general/FilenameBear.py (hunk 7 above)
Comment (Makman2) [inline bears/general/FilenameBear.py]: so you would insert something like this:
```
if not filename_without_extension.startswith(prefix):
    # modify new_name accordingly how it should be.

# same with suffix
```
Then you don't have to change any code in the result, except the message handling. Actually you can append messages inside a message list:

```
if not filename_without_extension.startswith(prefix):
    # modify new_name accordingly how it should be.
    messages.append('my message')
```
Then you don't need `itertools.compress` btw :)

File: bears/general/FilenameBear.py [hunk 9]
```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
...
+                            filename_prefix),
+                        not filename_without_extension.endswith(
+                            filename_suffix)]
 
             yield Result(
                 self,
-                'Filename does not follow {} naming-convention.'.format(
-                    file_naming_convention),
+                '\n'.join(
+                    itertools.compress(FilenameBear_messages, selectors)),
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: I'm maybe thinking of prepending bullets to each message. Instead of
```
message1
message2
...
```
-->
```
- message1
- message2
- ...
```
The `SpaceConsistencyBear` does it like this :) (Okay I have to admit that I've written that code :3)

Comment (Makman2) [inline bears/general/FilenameBear.py]: What do you think? :)

Comment (Makman2) [inline bears/general/FilenameBear.py]: Oh maybe we can be super smart and have an exception if there's only a single message:
```
if len(messages) == 1:
    # no bullet, just the message
else:
    # use bullets together with messages
```

File: bears/general/FilenameBear.py (hunk 4 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: I thought that `messages` isn't that clear.

File: bears/general/FilenameBear.py (hunk 5 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: I tried to follow the formatting that is already used in the file (Line 83).
https://github.com/coala/coala-bears/commit/9c4d2ed9295e8d067f7d60fe7210ae6344f4b027#diff-8da09498fa5dc8583a01d5e2f76de875L79

File: bears/general/FilenameBear.py (hunk 7 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Actually, this is much better :+1:  :smiley:

File: bears/general/FilenameBear.py (hunk 9 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: I think that ,after adding the bullets, the output looks better to me.
Special thanks goes to the `SpaceConsistencyBear`'s author :wink:.

File: bears/general/FilenameBear.py (hunk 8 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Well, i would like to work on it.
However, fulfilling the achievement that my PR is merged pushes me to say that i prefer working on it as a part of a new issue :thinking:

File: bears/general/FilenameBear.py [hunk 10]
```diff
@@ -74,12 +82,38 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        messages = []
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: Up to you:
I have the feeling that this belongs before the part with
```
try:
    new_name = self._naming_conventions[...]
```
as there we start to analyze the filename.

But like said, up to you, as semantically the current code is totally correct.

File: bears/general/FilenameBear.py [hunk 11]
```diff
@@ -74,12 +82,38 @@ def run(self, filename, file,
...
+                    self,
+                    '\n'.join('- ' + message for message in messages),
+                    diff.affected_code(filename),
+                    diffs={filename: diff})
+            else:
+                yield Result(
+                    self,
+                    messages[0],
+                    diff.affected_code(filename),
+                    diffs={filename: diff})
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: don't generate 2 results, but two different messages :D
```
if len(messages) > 1:
    result_message = '\n'.join('- ' + message for message in messages)
else:
    result_message = messages[0]

yield Result(
    self, result_message, diff.affected_code(filename), diffs={filename:diff})
```
or you can compact the `if`-statement to an `if`-expression if you like:
```
result_message = ('\n'.join('- ' + message for message in messages)
                  if len(messages) > 1 else
                  messages[0])
```

File: bears/general/FilenameBear.py [hunk 12]
```diff
@@ -74,12 +82,38 @@ def run(self, filename, file,
...
+        if not filename_without_extension.startswith(filename_prefix):
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the {} prefix.'.format(
+                    filename_prefix))
+
+        if not filename_without_extension.endswith(filename_suffix):
+            new_name = new_name + filename_suffix
+            messages.append(
+                'Filename does not use the {} suffix.'.format(
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: maybe displaying the `repr`? -->
`Filename does not use the {!r} suffix.`

What do you think?

Comment (Makman2) [inline bears/general/FilenameBear.py]: Same above^^

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Yes, i agree.
The output messages look much better.

File: bears/general/FilenameBear.py [hunk 13]
```diff
@@ -66,6 +74,24 @@ def run(self, filename, file,
...
+                new_name = filename_prefix + new_name
+                messages.append(
+                    'Filename does not use the {!r} prefix.'.format(
+                        filename_prefix))
+
+            if not filename_without_extension.endswith(filename_suffix):
+                new_name = new_name + filename_suffix
+                messages.append(
+                    'Filename does not use the {!r} suffix.'.format(
+                        filename_suffix))
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: ah no I didn't mean to put the whole stuff into the try-block, just moving `messages = []` *before* the try-block :D

The prefix/suffix stuff should come after the `try`-block :+1:

So
```python
        messages = []

        try:
            new_name = self._naming_convention[file_naming_convention](
                filename_without_extension)
        except KeyError:
            self.err('Invalid file-naming-convention provided: ' +
                     file_naming_convention)
            return

        if new_name != filename_without_extension:
            messages.append(
                'Filename does not follow {} naming-convention.'.format(
                    file_naming_convention))

        if not filename_without_extension.startswith(filename_prefix):
            new_name = filename_prefix + new_name
            messages.append(
                'Filename does not use the {!r} prefix.'.format(
                    filename_prefix))

        if not filename_without_extension.endswith(filename_suffix):
            new_name = new_name + filename_suffix
            messages.append(
                'Filename does not use the {!r} suffix.'.format(
                    filename_suffix))
```

Comment (Makman2) [inline bears/general/FilenameBear.py]: `try`-blocks should be left as early as possible, as we want to fail fast on other errors and don't want them to slip through :)

File: bears/general/FilenameBear.py [hunk 14]
```diff
@@ -74,12 +100,13 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        if messages:
+            diff = Diff(file, rename=os.path.join(
+                head, new_name + extension))
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: is the linebreak necessary here? Would keep the old line without a linebreak:
```
diff = Diff(file, rename=os.path.join(head, new_name + extension))
```
One change less to review ;)

File: bears/general/FilenameBear.py (hunk 13 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Ok

File: bears/general/FilenameBear.py [hunk 15]
```diff
@@ -71,15 +81,32 @@ def run(self, filename, file,
...
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the {!r} prefix.'.format(
+                    filename_prefix))
+
+        if not filename_without_extension.endswith(filename_suffix):
+            new_name = new_name + filename_suffix
+            messages.append(
+                'Filename does not use the {!r} suffix.'.format(
+                        filename_suffix))
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: indentation ;)

File: bears/general/FilenameBear.py [hunk 16]
```diff
@@ -71,15 +81,32 @@ def run(self, filename, file,
...
 
+        if new_name != filename_without_extension:
+            messages.append(
+                'Filename does not follow {} naming-convention.'.format(
+                    file_naming_convention))
+
+        if not filename_without_extension.startswith(filename_prefix):
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the {!r} prefix.'.format(
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: maybe rephrase slightly to
```
Filename does not use the prefix {!r}
```
Currently it sounds a bit "this is one well-known prefix".

Comment (Makman2) [inline bears/general/FilenameBear.py]: Same below for suffix :)

Comment (Makman2) [inline bears/general/FilenameBear.py]: But up to you :+1:

File: bears/general/FilenameBear.py [hunk 17]
```diff
@@ -71,15 +81,32 @@ def run(self, filename, file,
...
-        if new_name != filename_without_extension:
+        if messages:
             diff = Diff(file, rename=os.path.join(head, new_name + extension))
 
             yield Result(
                 self,
-                'Filename does not follow {} naming-convention.'.format(
-                    file_naming_convention),
+                '\n'.join('- ' + message for message in messages)
+                if len(messages) > 1 else messages[0],
```
Comment (Makman2) [inline bears/general/FilenameBear.py]: optional:
maybe put that into a `message` variable before, as this expression is a bit more complex?
up to you :+1:

File: tests/general/FilenameBearTest.py [hunk 18]
```diff
@@ -118,3 +118,17 @@ def test_auto_file_naming_convention_warning(self):
...
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='prefilename.xyz', valid=True)
+
+    def test_file_suffix(self):
+        self.section['filename_suffix'] = 'fix'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='filenamesuffix.xyz', valid=True)
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: Hm I would like to see here tests with `check_results` :)

Comment (Makman2) [inline tests/general/FilenameBearTest.py]: Shouldn't be too hard ;D

File: bears/general/FilenameBear.py (hunk 15 above)
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: I was embarrassed that i still have indentation problems but i think this message is the reason why i indented this line.
```
bears/general/FilenameBear.py                                                                                                         
|  99| ················filename_suffix))
|    | [NORMAL] PycodestyleBear (E122):
|    | E122 continuation line missing indentation or outdented
|    | *0: Do nothing
|    |  1: Open file(s)
|    |  2: Add ignore comment

```

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: @Makman2 Could you please check the message?
I won't push my code until we can figure out the correct way to do it.

Comment (Makman2) [inline bears/general/FilenameBear.py]: can't reproduce, a run on coala locally on your PR yields no errors :)

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: You should first unindent the lines as requested to reproduce the error message.

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: Well, the error is reproducible on my local env.
Anyway, i will unindent lines 93 and 99 as requested and push the changes without running coala.
Let's see what will gitmate say :smile:

File: bears/general/FilenameBear.py [hunk 19]
```diff
@@ -71,15 +81,33 @@ def run(self, filename, file,
...
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the prefix {!r}.'.format(
+                filename_prefix))
+
+        if not filename_without_extension.endswith(filename_suffix):
+            new_name = new_name + filename_suffix
+            messages.append(
+                'Filename does not use the suffix {!r}.'.format(
+                filename_suffix))
```
Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: @Makman2 Kindly check the error.

Comment (Makman2) [inline bears/general/FilenameBear.py]: naaaaah you shall indent like this!
```
            messages.append(
                'Filename does not use the suffix {!r}.'.format(
                    filename_suffix))
```
:D

Comment (AMR-KELEG) [inline bears/general/FilenameBear.py]: @Makman2 I know how to fix it.
I just wanted to show you that the error reproducible and the lines shouldn't be unindented as requested.
Tell me if I am misunderstanding the situation :)

File: tests/general/FilenameBearTest.py [hunk 20]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
         self.assertEqual(log.message,
                          'The file naming convention could not be guessed. '
                          'Using the default "snake" naming convention.')
+
+    def test_file_prefix(self):
+        self.section['filename_prefix'] = 'pre'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: use `check_invalidity` instead :)

File: tests/general/FilenameBearTest.py [hunk 21]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
...
+        self.check_validity(
+            self.uut, [''], filename='prefilename.xyz', valid=True)
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: you can omit the parameter `valid` :+1:

File: tests/general/FilenameBearTest.py [hunk 22]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
...
+                                'Filename does not use '
+                                'the prefix {!r}.'.format('pre'),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
+
+    def test_file_suffix(self):
+        self.section['filename_suffix'] = 'fix'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: same here -> `check_invalidity`

File: tests/general/FilenameBearTest.py [hunk 23]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
...
+        self.check_validity(
+            self.uut, [''], filename='filenamesuffix.xyz', valid=True)
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: `valid=True` can be omitted

File: tests/general/FilenameBearTest.py [hunk 24]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
...
+        self.section['filename_prefix'] = 'pre'
+        self.section['filename_suffix'] = 'fix'
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                '\n'.join(['- Filename does not use '
+                                           'the prefix {!r}.'.format('pre'),
+                                           '- Filename does not use '
+                                           'the suffix {!r}.'.format('fix')]),
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: instead of copying the functionality with join in the algorithm, try to write the string out:
```
"- Filename does not use the prefix 'pre'\n"
"- Filename does not use the suffix 'fix'"
```
So we directly see what we shall get :)
You could do the same above for the `.format` function uses above in the test.

File: tests/general/FilenameBearTest.py [hunk 25]
```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
...
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
```
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: Do you assert for the diffs? That would be cool to test too :)

File: bears/general/FilenameBear.py [hunk 26]
```diff
@@ -63,6 +71,8 @@ def run(self, filename, file,
                           'Using the default "snake" naming convention.')
                 file_naming_convention = 'snake'
 
+        messages = []
```
Comment (meetmangukiya) [inline bears/general/FilenameBear.py]: I don't like the idea of collecting all the errors into one diff, what if I want to apply the patch that renames it to follow naming convention, but not the patch that renames it with prefix text :confused: ?

Comment (meetmangukiya) [inline bears/general/FilenameBear.py]: Please consider yielding 3 results instead

File: tests/general/FilenameBearTest.py (hunk 25 above)
Comment (AMR-KELEG) [inline tests/general/FilenameBearTest.py]: Could you please provide another test file that uses diffs?

File: bears/general/FilenameBear.py (hunk 26 above)
Comment (Makman2) [inline bears/general/FilenameBear.py]: Nope this is on purpose, so we can generate a single patch without conflict errors :+1:

Comment (Makman2) [inline bears/general/FilenameBear.py]: If we support merging diffs on a file-name basis, we can yield again different results for those :+1:

File: tests/general/FilenameBearTest.py (hunk 25 above)
Comment (Makman2) [inline tests/general/FilenameBearTest.py]: Actually I was searching for one, and discovered no tests use `diffs`. I was wondering why your tests did already pass, because you definitely generate results with diffs. I found out that `Result` doesn't take diffs into account when comparing with each other. This is super bad and needs definitely a fix. Filed an issue here: https://github.com/coala/coala/issues/4400

Because of this, you don't have to provide a diff as it's meaningless now (via the `diffs` parameter of `Result.__init__`), but we file an issue once we are able to do so :+1:

**Comment by AMR-KELEG** [comment]
@Makman2 I know the organisation is reviewing the GSoC projects now.
In the meanwhile, could you tell me how can i fix the ci/circleci failing problem.
Should i re-push my commit to the branch?
Thanks :smile:

**Comment by Makman2** [comment]
ack f05e746

**Comment by Makman2** [comment]
@AMR-KELEG that is a known issue that is caused by `language-check`. The error is occurring randomly, I restarted the build :+1:

**Comment by AMR-KELEG** [comment]
@Makman2 thanks 😄 
I really enjoyed the whole review process and I am looking forward to collaborating with you in further contributions to coala.

**Comment by AMR-KELEG** [comment]
@Makman2 I have rebased my branch yet ci failed again :cry:

**Comment by Makman2** [comment]
> `bears/documentation/DocumentationStyleBear.py                             42      5     18      1    87%   103-116, 49->exit

looks like it's not your fault. I hope that's not happening also on master...

**Comment by Makman2** [comment]
@AMR-KELEG I hope a rebase does it now, as other PRs do pass :+1:

**Comment by jayvdb** [comment]
ack 404db94

**Comment by jayvdb** [comment]
@rultor merge
